### DISCUSSION TITLE
Indicate Firefox PiP MiniPlayer

### DISCUSSION THREAD

**Discussion by Riajyuu** [File-New-Project/EarTrumpet#698] [Ideas]
Context:
Since v71 Desktop, Firefox introduced a PiP UI. 
Sample like this:
![image](https://user-images.githubusercontent.com/16912240/71454176-6d893c00-2787-11ea-9ea9-d061549a7c43.png)

Expected behaviors: 
Use a special indicator to imply this is PiP mini-player instead of any other processes.

Why not use the animation effect?
Because it can be other tabs playing audios, like online music instead of PiP mini-player. I can tell here only because it is only thing playing sounds.

Why not work by Mozilla adding audio slide bar?
They seems not making this happen. See https://bugzilla.mozilla.org/show_bug.cgi?id=1580095

Plausible solution:
Use the `Picture-in-Picture` window title

**Comment by AlexWayfer** [comment]
> Plausible solution:
> Use the Picture-in-Picture window title

Also maybe Firefox can use a different icon for PiP player.

**Comment by Riajyuu** [comment]
> Also maybe Firefox can use a different icon for PiP player.

I believe they already did :-)

![image](https://user-images.githubusercontent.com/16912240/71506892-9c340f00-287a-11ea-98c7-1392b00b2b37.png)

**Comment by riverar** [comment]
Went through the Moz codebase. Picture-in-Picture is simply another Firefox window without chrome and without distinguishing features, other than a localized title. EarTrumpet needs a way to tap into Firefox and retrieve window attributes.

Related #448

**Comment by Riajyuu** [comment]
@riverar 
> needs a way to tap into Firefox and retrieve window attributes.

I believe this means a Firefox extension.
See https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging
Though similar tech can work for Chrome too.
